gk/main.c

```diff
@@ -1995,15 +1983,12 @@ gk_proc(void *arg)
 
 		process_cmds_from_mailbox(instance, gk_conf);
 
-		now = rte_rdtsc();
-		if (now - last_scan_tsc >= bucket_scan_timeout_cycles) {
+		if (iter_count % scan_iter == 0) {
```

Comment:
It would be better (1) to test `iter_count >= scan_iter`, (2) to set `iter_count` to zero inside of this `if` statements, (3) to add an `else` clause to increment `iter_count`, and (4) initialize `scan_iter` to just `gk_conf->flow_table_scan_iter` (no need for the plus one). I have two not-so-important reasons to do this:
1. The largest value `iter_count` can hold is `2^32−1 = 4294967295`. When `scan_iter` is 1001, the default value, the remainder above is `4294967295 % 1001 == 619`. Then `iter_count` is incremented, overflows, goes to zero and the remainder is zero again, that is, the loop skips all the iterations from 620 to 1000.
2. testing `iter_count >= scan_iter` is cheaper than `iter_count % scan_iter == 0`

Reply:
We certainly want this patch because varying the size of flow tables no longer affects performance. Thus, please update the documentation on our wiki.

Reply:
There's a third and **important** reason to follow my suggestion above, the remainder is undefined when `scan_iter` is zero.